packages/two_dimensional_scrollables/lib/src/table_view/table.dart

```diff
@@ -0,0 +1,1029 @@
+          cells: cells,
+          columnBuilder: columnBuilder,
+          rowBuilder: rowBuilder,
+        ));
+
+  @override
+  Widget buildViewport(BuildContext context, ViewportOffset verticalOffset,
+      ViewportOffset horizontalOffset) {
+    return TableViewport(
+      verticalOffset: verticalOffset,
+      verticalAxisDirection: verticalDetails.direction,
+      horizontalOffset: horizontalOffset,
+      horizontalAxisDirection: horizontalDetails.direction,
+      delegate: delegate,
+      mainAxis: mainAxis,
+      cacheExtent: cacheExtent,
+      clipBehavior: clipBehavior,
+    );
+  }
+}
+
+/// A widget through which a portion of a Table of [TableViewCell]s are viewed,
+/// typically in combination with a [TableView].
+class TableViewport extends TwoDimensionalViewport {
+  /// Creates a viewport for [TableViewCell]s that extend and scroll in both
+  /// horizontal and vertical dimensions.
+  const TableViewport({
+    super.key,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.delegate,
+    required super.mainAxis,
+    super.cacheExtent,
+    super.clipBehavior,
+  });
+
+  @override
+  TableCellDelegateMixin get delegate =>
+      super.delegate as TableCellDelegateMixin;
+
+  @override
+  RenderTwoDimensionalViewport createRenderObject(BuildContext context) {
+    return RenderTableViewport(
+      horizontalOffset: horizontalOffset,
+      horizontalAxisDirection: horizontalAxisDirection,
+      verticalOffset: verticalOffset,
+      verticalAxisDirection: verticalAxisDirection,
+      mainAxis: mainAxis,
+      cacheExtent: cacheExtent,
+      clipBehavior: clipBehavior,
+      delegate: delegate,
+      childManager: context as TwoDimensionalChildManager,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, RenderTableViewport renderObject) {
+    renderObject
+      ..horizontalOffset = horizontalOffset
+      ..verticalOffset = verticalOffset
+      ..cacheExtent = cacheExtent
+      ..clipBehavior = clipBehavior
+      ..delegate = delegate;
+  }
+}
+
+/// A render object for viewing [RenderBox]es in a table format that extends in
+/// both the horizontal and vertical dimensions.
+///
+/// [RenderTableViewport] is the visual workhorse of the [TableView]. It
+/// displays a subset of its children according to its own dimensions and the
+/// given [verticalOffset] and [horizontalOffset]. As the offset varies,
+/// different children are visible through the viewport.
+class RenderTableViewport extends RenderTwoDimensionalViewport {
+  /// Creates a viewport for [RenderBox] objects in a table format of rows and
+  /// columns.
+  RenderTableViewport({
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required super.delegate,
```

Comment:
Let's narrow down the type here do the `_delegate` cast below cannot fail.

```suggestion
    required TableCellDelegateMixin super.delegate,
```